mm2src/coins/eth/eth_swap_v2/eth_taker_swap_v2.rs

```diff
@@ -457,37 +458,120 @@ impl EthCoin {
+            invalid_token => {
+                return MmError::err(FindPaymentSpendError::InvalidData(format!(
+                    "Expected Token::FixedBytes, got {:?}",
+                    invalid_token
+                )))
+            },
+        };
+        // loop to find maker's spendTakerPayment transaction
         loop {
-            let taker_status = self
-                .payment_status_v2(
-                    taker_swap_v2_contract,
-                    decoded[0].clone(), // id from spendTakerPayment
-                    &TAKER_SWAP_V2,
-                    EthPaymentType::TakerPayments,
-                    TAKER_PAYMENT_STATE_INDEX,
-                )
-                .await?;
-            if taker_status == U256::from(TakerPaymentStateV2::MakerSpent as u8) {
-                return Ok(taker_payment.clone());
-            }
             let now = now_sec();
             if now > wait_until {
-                return MmError::err(WaitForPaymentSpendError::Timeout { wait_until, now });
+                return MmError::err(FindPaymentSpendError::Timeout { wait_until, now });
             }
-            Timer::sleep(10.).await;
+
+            let current_block = match self.current_block().compat().await {
+                Ok(b) => b,
+                Err(e) => {
+                    error!("Error getting block number: {}", e);
+                    Timer::sleep(5.).await;
+                    continue;
+                },
+            };
+
+            // get all logged TakerPaymentSpent events from `from_block` till current block
+            let events = match self
+                .events_from_block(taker_swap_v2_contract, "TakerPaymentSpent", from_block, current_block)
+                .await
+            {
+                Ok(events) => events,
+                Err(e) => {
+                    error!(
+                        "Error getting TakerPaymentSpent events from {} block: {}",
+                        from_block, e
+                    );
+                    Timer::sleep(5.).await;
+                    continue;
+                },
+            };
+
+            if events.is_empty() {
+                info!(
+                    "No events found yet for the block range {} to {}",
+                    from_block, current_block
+                );
+                Timer::sleep(5.).await;
+                continue;
+            }
+
+            // this is how spent event looks like in EtomicSwapTakerV2: event TakerPaymentSpent(bytes32 id, bytes32 secret)
+            let found_event = events.into_iter().find(|event| &event.data.0[..32] == id.as_slice());
+
+            if let Some(event) = found_event {
+                if let Some(tx_hash) = event.transaction_hash {
+                    let transaction = match self.transaction(TransactionId::Hash(tx_hash)).await {
+                        Ok(Some(t)) => t,
+                        Ok(None) => {
+                            info!("Tx {} not found yet", tx_hash);
```

Comment:
Could the log message be more informative?
`TakerPaymentSpendTx {} not found yet` for reading logs more easier
And maybe for this msg debug (not info) would suffice?

Reply:
> And maybe for this msg debug (not info) would suffice?

hmm using release binary you have to specify `RUST_LOG=debug` to see debug level logs, while ` info!("Tx {} not found yet", tx_hash);` notifies user that tx hash was found, but transaction not yet. 
I think its an important message and we shouldn't move it to debug level.

Reply:
updated log message a bit https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/fb383fb5556feda04cc0f0c841b64c5b64b051d2